### DISCUSSION TITLE
What is the correct way to import IEventAggregator from RegisterTypes?

### DISCUSSION THREAD

**Discussion by jamesnet214** [PrismLibrary/Prism#2781] [WPF] [answered]
I'm trying to provide the `IEventAggregator` injected into the `ViewModel` as a field inside the new instance object. So I got the `IEventAggregator` via `Container.Resolve<IEventAggregator>()`. Any problems with this method?

_Application_
```
protected override void RegisterTypes(IContainerRegistry containerRegistry)
{
    containerRegistry.RegisterInstance<MyEvent>(new MyEvent(Container.Resolve<IEventAggregator>()));
}
```
_ViewModel_
```
public void MainViewModel(MyEvent myEvent)
{

}
```

**Comment by jamesnet214** [comment]
Or should I use the ContainerLocator? Or is it okay if I choose anything because they're all the same?

**Comment by brianlagunas** [comment, marked as answer]
You don't need to do any of that. Just inject it in the VM ctor
```
public void MainViewModel(IEventAggregator ea)
{

}
```

Check out the doc topic: https://prismlibrary.com/docs/event-aggregator.html

**Reply by jamesnet214** [reply]
@brianlagunas I am honored to hear your reply.

Thank you.
